test/sync/src/androidTest/kotlin/io/realm/test/mongodb/shared/SyncConfigTests.kt

```diff
@@ -242,25 +241,74 @@ class SyncConfigTests {
         assertEquals(config.partitionValue.asString(), partitionValue)
     }
 
-//    @Test
-//    fun name() {
-//        val user: User = createTestUser(app)
-//        val filename = "my-file-name.realm"
-//        val config: SyncConfiguration = SyncConfiguration.Builder(user, DEFAULT_PARTITION)
-//            .name(filename)
-//            .build()
-//        val suffix = "/mongodb-realm/${user.app.configuration.appId}/${user.id}/$filename"
-//        assertTrue(config.path.endsWith(suffix))
-//    }
-//
-//    @Test
-//    fun name_illegalValuesThrows() {
-//        val user: User = createTestUser(app)
-//        val builder = SyncConfiguration.Builder(user, DEFAULT_PARTITION)
-//
-//        assertFailsWith<IllegalArgumentException> { builder.name(TestHelper.getNull()) }
-//        assertFailsWith<IllegalArgumentException> { builder.name(".realm") }
-//    }
+    @Test
+    fun name_withoutFileExtension() {
```

Comment:
We are missing a test for what the default value would look like.

Reply:
I added `name_similarToDefaultObjectStoreName` below, if that's what you mean?

Reply:
Yes, but that also passes in a user-defined name here: https://github.com/realm/realm-kotlin/pull/800/files/2c8370b257770e55e75b92ee8e731b6afb380939#diff-f3172b56e038676ca6061b7181e7eaae31885fe86018ccc332e00d2274b3524dR297

We are missing a test for what happens if no name is given.

Reply:
All the other tests in the suite that are not name-related don't specify a name so that should already be covered, no? On top of that, if Object Store assigns somewhat "random" names internally, to which don't have access until we instantiate the config object itself, what is it that we should test? I don't see how to compare the resulting name to an expected value.

Reply:
But we do know....partition values would be encoded with something like `s_<value>.realm` for Strings `i_<value>.realm` for Ints

The reason we need would to be ensure it was mostly for testing that the suffix `.realm` appear for these as well.